### PR TITLE
mp_capable: test MIBS for TCP fallback scenarios

### PR DESCRIPTION
**Pull request by dcaratti**
follow-up of upstream commit:

https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net.git/commit/?id=7a1b3490f47e88ec4cbde65f1a77a0f4bc972282

### CODE REVIEW

File: gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_flagH.pkt [hunk 1]
```diff
@@ -5,13 +5,18 @@
...
++0     `nstat -n`
 +0     bind(3, ..., ...) = 0
 +0     listen(3, 1) = 0
 +0       <  S   0:0(0)         win 32792  <mss 1460, sackOK, nop, nop, nop, wscale 8, mpcapable v1 flags 0x0 nokey>
 +0       >  S.  0:0(0)  ack 1             <mss 1460, nop, nop, sackOK, nop, wscale 8>
 +0.01    <   .  1:1(0)  ack 1  win 257
 +0     accept(3, ..., ...) = 4
 
+//  test mibs
++0     `test $(nstat -zjs | jq '.kernel.MPTcpExtMPCapableSYNRX') -eq 0`
```
Comment (matttbe) [inline gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_flagH.pkt]: maybe safer to use `+0.01` here as well?

File: gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_flagB_3rd_ack.pkt [hunk 2]
```diff
@@ -6,12 +6,17 @@
 +0     setsockopt(3, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
 
 +0     bind(3, ..., ...) = 0
++0     `nstat -n`
```
Comment (matttbe) [inline gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_flagB_3rd_ack.pkt:8]: detail: move that before the `bind` like the others?

File: gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_wrongver_3rd_ack.pkt [hunk 3]
```diff
@@ -5,13 +5,18 @@
...
 +0       <  S   0:0(0)         win 32792  <mss 1000, sackOK, nop, nop, nop, wscale 7, mpcapable v1 flags[flag_h] nokey>
 +0       >  S.  0:0(0)  ack 1             <mss 1460, nop, nop, sackOK, nop, wscale 8, mpcapable v1 flags[flag_h] key[skey]>
 +0.01    <   .  1:1(0)  ack 1  win 257                                               <mpcapable v0 flags[flag_h] key[ckey=2, skey]>
 +0     accept(3, ..., ...) = 4
 
+//  test mibs
++0.01  `test $(nstat -zjs | jq '.kernel.MPTcpExtMPCapableSYNRX') -eq 1`
++0     `test $(nstat -zjs | jq '.kernel.MPTcpExtMPCapableFallbackACK') -eq 1`
+
 // ensure that traffic plane is functional and does not use DSS
```
Comment (matttbe) [inline gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_wrongver_3rd_ack.pkt:20]: should you eventually add the same test in `dss/mpc_with_data_server.pkt`? We have a SYN + MPC, but no fallback. (even if it is covered by the selftests)

File: gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_nompc.pkt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
++0       <  S   0:0(0)         win 32792  <mss 1460, sackOK, nop, nop, nop, wscale 8>
++0       >  S.  0:0(0)  ack 1             <mss 1460, nop, nop, sackOK, nop, wscale 8>
++0.01    <   .  1:1(0)  ack 1  win 257
++0     accept(3, ..., ...) = 4
+
+//  test mibs
++0.01  `test $(nstat -zjs | jq '.kernel.MPTcpExtMPCapableSYNRX') -eq 0`
++0     `test $(nstat -zjs | jq '.kernel.MPTcpExtMPCapableFallbackACK') -eq 0`
+
+// ensure that traffic plane is functional and does not use DSS
```
Comment (matttbe) [inline gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_nompc.pkt:20]: not sure we need this, but it doesn't hurt!

**Review by matttbe — COMMENTED** [review]
thank you for having looked at this!

Just 2 small details, and a suggestion if that's OK for you

File: gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_wrongver_3rd_ack.pkt (hunk 3 above)
Comment (dcaratti) [reply gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_wrongver_3rd_ack.pkt:20]: so in this case we expect `MPTcpExtMPCapableFallbackACK = 0`  and `MPTcpExtMPCapableSYNRX=1`, correct?

Comment (matttbe) [reply gtests/net/mptcp/mp_capable/v1_bind_tcpfallback_wrongver_3rd_ack.pkt:20]: Yes, correct, it is just to check this "normal" case.

**Comment by matttbe** [comment]
Thank you for the new version!

**Pull request merged**